### PR TITLE
[Fix #258] Return JS candidates only for non-Cljs namespaces

### LINKED ISSUE #258: JS completions are too eager
e.g.
```
cljs.user=> (s/ends
(s/ends-with?        (s/ends_with_QMARK_
```

we shouldn't expand JS completions if they're a CLJS namespace

### PR DESCRIPTION
**Pull request by arichiardi**

### CODE REVIEW

File: test/lumo/repl_tests.cljs [hunk 1]
```diff
@@ -61,7 +61,9 @@
                                              '{string clojure.string})]
         (is-contains-completion "str" "string/")
         (is-contains-completion "(str" "(string/")
-        (is-contains-completion "(set" "(set/" not))
+        (is-contains-completion "(set" "(set/" not)
+        (testing "LUMO-258"
+          (is-contains-completion "string/ends" "string/ends-with?")))
```
Comment (anmonteiro) [inline test/lumo/repl_tests.cljs]: This is not enough. We also want to make sure that `ends_with_QMARK_` isn't there. That's the problem

File: src/cljs/snapshot/lumo/repl.cljs [hunk 2]
```diff
@@ -1365,7 +1369,8 @@
       (into #{} (mapcat identity)
         [(completion-candidates-for-ns full-ns false)
          (completion-candidates-for-ns (add-macros-suffix full-ns) false)
-         (completion-candidates-for-js-sources full-ns)]))
+         (when-not (get-namespace full-ns)
```
Comment (anmonteiro) [inline src/cljs/snapshot/lumo/repl.cljs]: Let's move this check to the Closure candidates function please

File: test/lumo/repl_tests.cljs (hunk 1 above)
Comment (arichiardi) [reply test/lumo/repl_tests.cljs]: Ok

File: src/cljs/snapshot/lumo/repl.cljs (hunk 2 above)
Comment (arichiardi) [reply src/cljs/snapshot/lumo/repl.cljs]: I can do it, but this expresses better that we don't want JS completions for cljs namespaces IMHO. Unless there is something I am missing of course

Comment (anmonteiro) [reply src/cljs/snapshot/lumo/repl.cljs]: just looks a bit clunky here. I'd prefer it localized

Comment (arichiardi) [reply src/cljs/snapshot/lumo/repl.cljs]: Ok no probs, will move it

**Comment by arichiardi** [comment]
Addressed the comments.

**Comment by anmonteiro** [comment]
Thanks

**Comment by arichiardi** [comment]
Very weird it completed and got all green but it does not anymore now here. The test seems to fail. I am digging.

**Pull request merged**